### PR TITLE
Bugfix: Correctly calculate Overlay position when inspecting a node within iframe(s)

### PR DESCRIPTION
**Pull request by suchipi**
This fixes https://github.com/facebook/react-devtools/issues/443.

I didn't see any (automated) tests in the repo, so I have not added any. Should I have?

I created a test app with some nested iframes that can be used with the plain shell. To use it, build `test/nested-frames` using webpack and add `data-app-src="../../test/nested-frames/build/target.js"` to the plain shell's target iframe.

I have only tested this with the plain shell; I'm not sure how to test it as an extension.

I created some types for `Document` and `Window`. They only have the properties I cared about for this diff on them right now, but could be fleshed out eventually if desired. There might also be a better set of type definitions for the DOM API out there somewhere that could be used instead.

### CODE REVIEW

File: test/nested-frames/target.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/**
```
Comment (gaearon) [inline test/nested-frames/target.js]: Can we add this example to the `kitchensink` instead?

Comment (suchipi) [inline test/nested-frames/target.js]: Sure... do you mean`test/example/sink.js`?

Comment (gaearon) [inline test/nested-frames/target.js]: Yep!

Comment (gaearon) [inline test/nested-frames/target.js]: It's easier to have them all in one place rather than create a new test for every case.

Comment (suchipi) [inline test/nested-frames/target.js]: Yup, that makes sense. I'll move it over there this evening

**Comment by suchipi** [comment]
Moved the example to sink and rebased on master. Not sure if the build output for sink should be committed, so I left that out.

**Comment by gaearon** [comment]
I did some testing, and it seems to work. Going to trust you on this one 😅 .

**Comment by suchipi** [comment]
Feel free to pull me back in if it breaks in the future 😉 

Thanks!

**Comment by rchanou** [comment]
When can we expect to see this version released to the chrome store? I imagine a lot of people using react-storybook want this.

**Comment by gaearon** [comment]
Can you ping me about it again next week?

**Comment by rchanou** [comment]
Will do!

**Comment by rchanou** [comment]
@gaearon Pinging as promised!

**Comment by gaearon** [comment]
Thanks! I'm waiting for a few more changes in other PRs so I can all of this together.

**Comment by suchipi** [comment]
@gaearon Any updates?

**Comment by gaearon** [comment]
Not currently. I remember about releasing the DevTools. Right now it's secondary to our sprint to get React built with Rollup. This is a giant change internally, and we're very focused on it, but it won't take more than about a week more of my time. Then I'll be able to look at this again and cut a release.

**Comment by suchipi** [comment]
Okay. Thanks for the info!

**Comment by suchipi** [comment]
Looks like rollup work is done (https://github.com/facebook/react/pull/9327); any chance this could be released?

**Comment by gaearon** [comment]
Rollup work also involves updating our FB scripts to build Rollup bundle, releasing an alpha for RN, and then updating React Native to remove internal dependencies and then use the flat bundle. I’m sorry it’s taking time, but I don’t want to leave it unfinished.

**Comment by suchipi** [comment]
Is there anything the community can do to help get this released sooner? As @rchanou indicated, this bugfix is significant for react storybook users (like myself).

**Comment by gaearon** [comment]
Would you like to help with reviewing and merging pending PRs that are (close to) approved?

**Comment by suchipi** [comment]
Sure; I should have some time to do that this Saturday.

**Comment by gaearon** [comment]
@suchipi Apologies for late reply, took a while to approve the request in our internal systems 😞 You should have commit access now so please feel free to review and merge PRs you think are stable. I suggest starting with https://github.com/facebook/react-devtools/pull/459, https://github.com/facebook/react-devtools/pull/541, and https://github.com/facebook/react-devtools/pull/569. Please make sure to test changes extensively.

**Comment by suchipi** [comment]
Thanks @gaearon. I will probably not have time to look at these until the upcoming weekend, but I took a look at a few other PRs in the meantime yesterday.

**Comment by gaearon** [comment]
Thanks!

**Comment by suchipi** [comment]
Sorry I didn't look at those PRs last weekend @gaearon. Hopefully I will have some time this weekend

**Pull request merged**